application/classes/Ushahidi/Repository/Post/Lock.php

```diff
@@ -0,0 +1,111 @@
+<?php defined('SYSPATH') OR die('No direct access allowed.');
+
+/**
+ * Ushahidi Post Lock Repository
+ *
+ * @author     Ushahidi Team <team@ushahidi.com>
+ * @package    Ushahidi\Application
+ * @copyright  2017 Ushahidi
+ * @license    https://www.gnu.org/licenses/agpl-3.0.html GNU Affero General Public License Version 3 (AGPL3)
+ */
+
+use Ushahidi\Core\Entity;
+use Ushahidi\Core\Entity\PostLock;
+use Ushahidi\Core\Entity\PostLockRepository;
+use Ushahidi\Core\Traits\UserContext;
+
+class Ushahidi_Repository_Post_Lock extends Ushahidi_Repository implements PostLockRepository
+{
+	// Provides getUser()
+	use UserContext;
+
+	// Ushahidi_Repository
+	protected function getTable()
+	{
+		return 'post_locks';
+	}
+
+	// Ushahidi_Repository
+	public function getSearchFields()
+	{
+		return [
+			'post_id',
+			'user_id'
+		];
+	}
+
+    // Ushahidi_Repository
+	public function getEntity(Array $data = null)
+	{
+		return new PostLock($data);
+	}
+
+	public function releaseLock($entity_id)
+	{
+		$query = DB::delete('post_locks')
+			->where('post_id', '=', $entity_id);
+		return $query->execute();
+	}
+
+	public function releaseLockByLockId($lock_id)
+	{
+		$query = DB::delete('post_locks')
+			->where('id', '=', $lock_id);
+		return $query->execute();
+	}
+
+	public function checkLock($entity_id)
```

Comment:
isActive or isExpired might make it easier to read. $this->isActive($is) vs !$thick->checkLock($id)

Reply:
At the same time, all the other methods in the class use doSomething`Lock`() instead of doSomething() so maybe it makes sense to name is isLockActive instead of isActive. Usually I prefer ->get vs getLock in this scenario, or ->create vs ->createLock unless we were adding that method on the post repository for some reason.

Reply:
Excellent notes, I'll review the naming.

Reply:
for get, there is a default get method applied from a parent so I'm going to keep getLock to differentiate

Reply:
Makes sense 👍